hazelcast/src/main/java/com/hazelcast/map/impl/record/AbstractRecordWithStats.java

```diff
@@ -38,7 +40,9 @@ public final RecordStatistics getStatistics() {
     }
 
     public final void setStatistics(RecordStatistics recordStatistics) {
-        this.recordStatistics = recordStatistics;
+        if (recordStatistics != null) {
+            this.recordStatistics = recordStatistics;
+        }
```

Comment:
`else` assign a `new RecordStatistics()` to it, perhaps?  Seems like a setter really should set something if called with a `null`.

If the Hazelcast code was a bit more consistent with nulls, it might make more sense to catch this at a higher layer, so that all this method should need to do is `requireNonNull` or whatever.  But that would be a decision for the Hazelcast folks to make.